pio-scripts/set_repo.py

```
@@ -0,0 +1,54 @@
+    except FileNotFoundError:
+        # Git CLI is not installed or not in PATH
+        return 'unknown'
+    except subprocess.CalledProcessError:
+        # Git command failed (e.g., not a git repo, no remote, etc.)
+        return 'unknown'
+    except Exception:
+        # Any other unexpected error
+        return 'unknown'
+
+repo = get_github_repo()
+env.Append(BUILD_FLAGS=[f'-DWLED_REPO=\\"{repo}\\"'])
```

Ruff (0.14.10)

Undefined name `env`

(F821)